presto-main/src/main/java/com/facebook/presto/sql/analyzer/FeaturesConfig.java

```diff
@@ -282,6 +282,8 @@ public class FeaturesConfig
 
     private boolean preProcessMetadataCalls;
 
+    private boolean removeRedundantCastToVarcharInJoin = true;
```

Comment:
why do we need a session property? isn't this always beneficial?

Reply:
Just in case we need to revert it if any bugs found later.